### PR TITLE
Added Teensy 4.1 mac address getter to Ethernet publisher example

### PR DESCRIPTION
**Pull request by adityakamath**
Signed-off-by: Aditya Kamath <adityakamath@live.com>

### CODE REVIEW

**Review by pablogs9 — APPROVED** [review]

**Comment by pablogs9** [comment]
@mergify backport foxy main humble

File: examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino [hunk 1]
```diff
@@ -22,30 +29,26 @@ rcl_node_t node;
...
-    RCSOFTCHECK(rcl_publish(&publisher, &msg, NULL));
-    msg.data++;
-  }
-}
-
 void setup() {
+  #if defined(ARDUINO_TEENSY41)
+  uint8_t mac[6];
+  get_teensy_mac(mac);
+  byte arduino_mac[] = { mac[0], mac[1], mac[2], mac[3], mac[4], mac[5] };
```
Comment (pablogs9) [inline examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino:43]:
```suggestion
  uint8_t arduino_mac[6];
  get_teensy_mac(arduino_mac);
```

File: examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino [hunk 2]
```diff
@@ -22,30 +29,26 @@ rcl_node_t node;
...
+  #else
   byte arduino_mac[] = { 0xAA, 0xBB, 0xCC, 0xEE, 0xDD, 0xFF };
+  #endif
```
Comment (pablogs9) [inline examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino:46]:
```suggestion
   byte arduino_mac[] = { 0xAA, 0xBB, 0xCC, 0xEE, 0xDD, 0xFF };

  #if defined(ARDUINO_TEENSY41)
  get_teensy_mac(arduino_mac);
  #endif
```

**Comment by pablogs9** [comment]
@Acuadros95 please review

**Review by Acuadros95 — APPROVED** [review]
LGTM

**Pull request merged**